client/helpers.go

```diff
@@ -302,7 +302,7 @@ func isNotFoundError(err error) bool {
 
 func IsInvalidParameterValueError(err error) bool {
 	var apiErr smithy.APIError
-	if errors.As(err, &apiErr); apiErr.ErrorCode() == "InvalidParameterValue" {
+	if ok := errors.As(err, &apiErr); ok && apiErr.ErrorCode() == "InvalidParameterValue" {
```

Comment:
```suggestion
	if errors.As(err, &apiErr) && apiErr.ErrorCode() == "InvalidParameterValue" {
```
simpler?

Reply:
yes this should be the way to go and the suggested one

Reply:
@yevgenypats You're missing that current code does not test a result of `errors.As`, so if it is `false` then `apiErr` is nil thus panic.